rest-service/manager_rest/blueprints_manager.py

```diff
@@ -856,6 +863,51 @@ def _get_only_user_execution_parameters(execution_parameters):
         return {k: v for k, v in execution_parameters.iteritems()
                 if not k.startswith('__')}
 
+    def _get_resolver_section(self):
+        resolver_class_path = None
+        params = None
+        provider_context = self.sm.get_provider_context().context
+        if self.sm.get_provider_context().context:
+            cloudify_section = provider_context['cloudify']
+            resolver_section = cloudify_section.get(constants.URL_RESOLVER_KEY)
+            if resolver_section:
+                resolver_class_path = resolver_section.get(
+                    constants.RESOLVER_IMPLEMENTATION_KEY)
+                params = resolver_section.get(
+                    constants.RESLOVER_PARAMETERS_KEY)
+        return resolver_class_path, params
+
+    def _update_url_resolver(self):
+        resolver = DefaultUrlResolver()
+        resolver_class_path, params = self._get_resolver_section()
+        if resolver_class_path:
+            try:
+                resolver = get_class_instance(resolver_class_path, params)
+            except Exception as e:
+                raise ResolverInstantiationError(e.message)
+        else:
+            # using the default resolver
+            if params:
+                # using custom rules for the default resolver
+                try:
+                    resolver = DefaultUrlResolver(
+                        rules=params.get(constants.DEFAULT_RESLOVER_RULES_KEY))
+                except Exception as e:
+                    raise ResolverInstantiationError(
+                        'failed to instantiate the default resolver ({0}) '
+                        'with params {1}: {2}'.format(
+                            DefaultUrlResolver.__class__.__name__,
+                            params, e.message))
+        return resolver
+
+    def update_provider_context(self, update, context):
+        if update:
+            self.sm.update_provider_context(context)
+        else:
+            self.sm.put_provider_context(context)
+
+        current_app.resolver = self._update_url_resolver()
```

Comment:
why the additional storage call? you already have `context` why not pass it directly to `_update_url_resolver` and the to `_get_resolver_section`?